src/main/scala/firrtl/passes/ExpandWhens.scala

```diff
@@ -61,13 +64,29 @@ object ExpandWhens extends Pass {
     * @note Seq[Attach] contains all Attach statements (unsimplified)
     * @note Statement contains all declarations in the module (including DefNode's)
     */
-  def expandWhens(m: Module): (Netlist, Simlist, Seq[Attach], Statement) = {
+  def expandWhens(m: Module): (Netlist, Simlist, Seq[Attach], Statement, InfoMap) = {
     val namespace = Namespace(m)
     val simlist = new Simlist
     val nodes = new NodeMap
     // Seq of attaches in order
     lazy val attaches = mutable.ArrayBuffer.empty[Attach]
 
+    val infoMap: InfoMap = new InfoMap {
+      override def default(key: String): Info = {
+        this(key) = NoInfo
+        this(key)
+      }
+    }
```

Comment:
Using .withDefaultValue makes it harder to pass around the type.
See [https://github.com/scala/bug/issues/8099](https://github.com/scala/bug/issues/8099)
There's probably a work-around but I haven't found it yet

Reply:
I'm not sure what you mean by "pass around the type", especially since you're adding this default to the instance `infoMap` rather than to the type `InfoMap`, and setting the type of `infoMap` to be the default-less `InfoMap` anyway.

But I do see how overriding `default` actually gives `Some(NoInfo)` upon a `.get(...)` so that's useful. I guess you generally shouldn't be using `.get` with a map with a default but I'm sure there are reasons one would want to. 

I would be surprised if the JIT doesn't figure it out, but it still might be better to simply return `NoInfo` instead of `this(key)`.

Reply:
When I used `val infoMap = (new InfoMap).withDefaultValue(NoInfo)`
infoMap was not an acceptable in the return signature of expandWhens `.., InfoMap)`
I've tried playing with it adding WithDefault[String, Info] but haven't found the magic combo

Reply:
Oh I see. Well since it's clearly implicit that all InfoMaps need to have a default value, I think you should just change the type `InfoMap`, in that instead of being a type alias, it should just subclass `HashMap` and the default should be put there.